### PR TITLE
(feat) : added a refund option to the bill manager.

### PR DESCRIPTION
**Pull request by amosmachora**
## Requirements

- [x] This PR has a title that briefly describes the work done, including the ticket number if there is a ticket.
- [x] My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide).
- [ ] I checked for feature overlap with [**existing widgets**](https://om.rs/directory).

## Summary
In this PR I have added a refund option to the bill manager. Now administrators could be able to refund line items. This means that refunded items will be consequentially not count towards the total of the bill.

Also the UI marks the refunded lineitem rows with a pale red for accountability purposes. This approach was chosen over filtering out the refunded items. Thanks @ojwanganto for the help.

I also added checks to make sure we do not refund already refunded items.

### Refund Option
image
### Screen With Refunded Items
image
### Refund Item Modal
image

## Related Issue

*None.*

## Other

*None.*

### CODE REVIEW

**Comment by ojwanganto** [comment]
Thanks, @amosmachora. Can we only shade credited row(s)?

**Comment by amosmachora** [comment]
The only rows shaded are the credited rows. Maybe the screenshot doesn't do the best job in showing that but I only shade the credited rows.

**Comment by ojwanganto** [comment]
image
From this, I see 2 rows for Malaria test. One has a positive figure and the other negative. Both rows are shaded.

**Comment by amosmachora** [comment]
Yeah it is by design. Remember we add a new lineitem with a negative value so that it gets subtracted from the total. Thats why both are shaded implying they are refunded items. I dont know maybe we should filter out one to avoid confusion?

**Comment by ojwanganto** [comment]
just shade the one with negative value.

**Comment by amosmachora** [comment]
Fixed it @ojwanganto

image

**Comment by ojwanganto** [comment]
Thanks, @amosmachora. CC @donaldkibet

File: packages/esm-billing-app/src/billable-services/bill-manager/bill-manager.scss [hunk 1]
```diff
@@ -12,3 +12,15 @@
 .dataTableSkeleton {
   margin-top: 0.625rem;
 }
+
+.refundedItem {
+  background-color: #fee2e2;
```
Comment (donaldkibet) [inline packages/esm-billing-app/src/billable-services/bill-manager/bill-manager.scss:17]: We should be using colors from carbon token system e.g

```scss
@use '@cabon/colors';
```

File: packages/esm-billing-app/src/billable-services/bill-manager/bill-manager.scss [hunk 2]
```diff
@@ -12,3 +12,15 @@
...
+}
+
+.refundedItem:hover {
+  background-color: #fecaca !important;
```
Comment (donaldkibet) [inline packages/esm-billing-app/src/billable-services/bill-manager/bill-manager.scss:21]: Using `!important` isn't recommend and used as the last resort. I would discourage its usage here

File: packages/esm-billing-app/src/billable-services/bill-manager/modals/cancel-bill.scss [hunk 3]
```diff
@@ -7,3 +7,13 @@
 .modalHeaderHeading {
   @include type.type-style('heading-compact-02');
 }
+
+.button_spinner {
+  padding: 0;
+  margin-right: 12px;
```
Comment (donaldkibet) [inline packages/esm-billing-app/src/billable-services/bill-manager/modals/cancel-bill.scss:13]: Lets start using `@carbon` tokens for spacing, typography to make maintaining this work easy.

**Review by donaldkibet — APPROVED** [review]
Thanks @amosmachora Am merging this please do a follow up PR to address the feedback.

File: packages/esm-billing-app/src/billable-services/bill-manager/bill-manager.scss (hunk 2 above)
Comment (amosmachora) [reply packages/esm-billing-app/src/billable-services/bill-manager/bill-manager.scss:21]: It was actually a last resort. I couldnt get anything else to work. But I did not try the `carbon/colors` option so maybe that will work. Thanks for the feedback will work on the rest of the feedback before the end of today.

**Pull request merged**